### PR TITLE
Support for pushing and pulling ai models to and from the model repository

### PR DESCRIPTION
**Pull request by marccampbell**

### CODE REVIEW

File: cli/cmd/model_pull.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	// get the endpoint
+	endpoint, err := r.kotsAPI.GetModelsEndpoint()
+	if err != nil {
+		return err
+	}
+
+	if endpoint == "" {
+		return errors.New("Could not find models endpoint")
+	}
+	fmt.Printf("Pulling model from %s\n", endpoint)
```
Comment (divolgin) [inline cli/cmd/model_pull.go:46]: Is there a progress display when oras.Copy is called?

File: pkg/ociclient/upload.go [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+	"net/http"
+	"os"
+	"time"
+
+	digest "github.com/opencontainers/go-digest"
+	specs "github.com/opencontainers/image-spec/specs-go"
+	v1 "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+const chunkSize = 100 * 1024 * 1024 // 100 MB chunks
```
Comment (divolgin) [inline pkg/ociclient/upload.go:20]: How about changing this to upload the whole file until it chunking works in the registry correctly?

**Review by divolgin — APPROVED** [review]

File: cli/cmd/model_pull.go (hunk 1 above)
Comment (marccampbell) [reply cli/cmd/model_pull.go:46]: not at this time

File: pkg/ociclient/upload.go (hunk 2 above)
Comment (marccampbell) [reply pkg/ociclient/upload.go:20]: agreed, this only works for 100 MB files.  but larger gets really slow also.  
i'd prefer to keep this chunked and not working for > 100 MB than unknown size limitations.
product is not yet in prod

**Pull request merged**